integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/CarbonAlterTableRenameCommand.scala

```diff
@@ -156,18 +156,21 @@ private[sql] case class CarbonAlterTableRenameCommand(
     } catch {
       case e: Exception =>
         LOGGER.error(e, "Rename table failed: " + e.getMessage)
-        if (carbonTable != null) {
-          AlterTableUtil
-            .revertRenameTableChanges(oldTableIdentifier,
+        try {
```

Comment:
Still it is unknown, when exception is raised before rename or after rename.
So better handle unlocking at common place and logic and change table path if rename is success.

Reply:
done